### ISSUE TITLE
ocrmypdf fails - it seems file extensions are mixed up

### ISSUE THREAD

**Issue by WillemJansen** [jbarlow83/OCRmyPDF#356]
Dear all,
I am  trying to get ocrmypdf v8.1.0 to run on my raspberry pi. I had to do a lot of compiling, but so far it seems to go in the right direction. Thus, this might be a problem with all the self-compiled libraries, but I suspect something much more mundane such as simple mix-up of filename extensions ...

**Describe the issue**
When I try to use OCRmyPDF on scanned files, the program returns an error as

`No way to run job: Input file '/tmp/com.github.ocrmypdf.xe4h16ff/000001.text.pdf' does not exist`
(Error message is correct as file does not exist. However, '000001.text' DOES!)

**To reproduce**
`TESSDATA_PREFIX=/usr/share/tesseract-ocr/tessdata ocrmypdf -l deu+eng  -j 2 -v 10 -k --tesseract-timeout 10000  --mask-barcodes --output-type pdf   20190224_112220.pdf Test12.pdf`

**Example file**
[20190224_112220.pdf](https://github.com/jbarlow83/OCRmyPDF/files/2897949/20190224_112220.pdf) attached

Please check any or all that apply about the test file:
- [x] This is the input file
- [x] The file contains no personal or confidential information
- [] I am the copyright holder for this file
- [] I permit this file to be included in the OCRmyPDF test suite under the CC-BY-SA 4.0 license
- [] I am not the copyright holder, but this file is available under a free software license

**Expected behavior**
OCRed lay on top of scanned PDF.

**System:**
- OS: Raspbian stretch
- OCRmyPDF Version: v8.1.0 

**Additional context**
[output.txt](https://github.com/jbarlow83/OCRmyPDF/files/2897954/output.txt)

Based on the detailed (-v 10) output logs, it seems that tesseract creates a file called 000001.text, for the subsequent steps, however, ocrmypdf seems to look for a file called 000001.text.pdf ..

Also attached: 
`ll /tmp/com.github.ocrmypdf.xe4h16ff`
[ContentsTempDir.txt](https://github.com/jbarlow83/OCRmyPDF/files/2897958/ContentsTempDir.txt)

**Comment by jbarlow83** [comment]
I suspect that pikepdf is broken somehow.

Can you run the pikepdf test suite and see if it passes?

**Comment by WillemJansen** [comment]
More than happy to test pikepdf ... unfortunately I am not sure, if I am doing it right?
`python3.7 setup.py test` fails completely,not at any tests, but at some import errors. Either I am doing it dead wrong of the answer is right there staring me in the face :)

**Comment by WillemJansen** [comment]
Okay. testing compelted, mostly successful (see pull request).

Just went a bit deeper into the logfiles:
`DEBUG - ['tesseract', '-l', 'deu+eng', '-c', 'textonly_pdf=1', '/tmp/com.github.ocrmypdf.pl2rjv8n/000002.ocr.png', '/tmp/com.github.ocrmypdf.pl2rjv8n/000002.text', 'pdf', 'txt']`
should translate to:
`tesseract -l deu+eng -c textonly_pdf=1 /tmp/com.github.ocrmypdf.pl2rjv8n/000002.ocr.png /tmp/com.github.ocrmypdf.pl2rjv8n/000002.text pdf txt`
right?

This would explain why tesseract fails. Moreover, when I run the line above in the shell, I get the exact same error message as in the attached output.

This leaves us with the Million Dollar Question: What does line 344 in tesseract.py do ... and what is the intention of the 'pdf', 'txt' parameters?

**Comment by jbarlow83** [comment]
```
  DEBUG - ['tesseract', '-l', 'deu+eng', '-c', 'textonly_pdf=1', '/tmp/com.github.ocrmypdf.xe4h16ff/000002.ocr.png', '/tmp/com.github.ocrmypdf.xe4h16ff/000002.text', 'pdf', 'txt']
  DEBUG - ['tesseract', '-l', 'deu+eng', '-c', 'textonly_pdf=1', '/tmp/com.github.ocrmypdf.xe4h16ff/000001.ocr.png', '/tmp/com.github.ocrmypdf.xe4h16ff/000001.text', 'pdf', 'txt']
  ERROR -    2: [tesseract] read_params_file: Can't open pdf
  ERROR -    2: [tesseract] read_params_file: Can't open txt
```

Do I get a million dollars for answering the million dollar question?

`pdf` and `txt` are the names of Tesseract configuration files to use when processing. The fact that they cannot be opened means Tesseract is not installed correctly.

The folder `<base>/share/tessdata/configs/` should contain plain text files named `pdf` and `txt`. `<base>` could be `/usr` or `/usr/local` or some other prefix, depending on the system.

You should be able to run a command like
```
tesseract -l deu+eng -c textonly_pdf=1 image.png output pdf txt
```

and it should generate `output.pdf` and `output.txt`. (Because of `-c textonly_pdf=1` the PDF will be a blank page that contains invisible text.)

**Comment by WillemJansen** [comment]
More than happy to give away the million dollars ... unfortunately, I am about a million dollars short for that :)

Just wanted to thank you a lot for you comments, I did indeed mess up the tesseract config: by pointing to the language files in the folder created by the old installation I broke the link to the propoer tessdata folder. I changes that and now it works like a charm even on the rapsberry pi (a slow charm, of course).

Thus, thanks a lot, really learned a lot in the process!

   Best,
    Willem
